infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/VerifyingRemoteSinkLeaderState.java

```diff
@@ -60,7 +61,10 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 fsm.updateConnectedNodes(event.getNodeId());
                 return this;
             case LOCAL_LEADER_LOSS:
-                return fsm.getStates().get(LogReplicationRuntimeStateType.STOPPED);
+                if (canEnqueueStopRuntimeFsmEvent(router, fsm, event.isConnectionStarter)) {
```

Comment:
minor:  This name and behavior is a bit confusing.  canEnqueueStopRuntimeFsmEvent() performs the check and sends leadership loss.  Whether connection starter or not, STOPPED event is enqueued.

Maybe the name and/or code can be reorganized to something like this:

```

if (!connectionStarter) {
       sendLeadershipLoss()
}
//enqueue the fsm event
```

Reply:
its more like, STOPPED event can be immediately enqueued only when the SOURCE is the connection starter.
Otherwise, we need to wait/resend leadership_loss request until an ACK is received. This is to ensure that SINK has received the msg, and that it will initiate broadcasting LEADERSHIP_QUERY.

So we need the logic we have currently in the canEnqueueStopRuntimeFsmEvent.